component/api/src/main/java/org/exoplatform/social/metadata/attachment/AttachmentService.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * This file is part of the Meeds project (https://meeds.io/).
+ *
+ * Copyright (C) 2023 Meeds Association contact@meeds.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.exoplatform.social.metadata.attachment;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.exoplatform.commons.exception.ObjectNotFoundException;
+import org.exoplatform.services.security.Identity;
+import org.exoplatform.services.security.IdentityRegistry;
+import org.exoplatform.social.metadata.AttachmentPlugin;
+import org.exoplatform.social.metadata.attachment.model.ObjectUploadResourceList;
+import org.exoplatform.social.metadata.attachment.model.ObjectAttachmentDetail;
+import org.exoplatform.social.metadata.attachment.model.ObjectAttachmentList;
+import org.exoplatform.social.metadata.attachment.model.ObjectAttachmentOperationReport;
+import org.exoplatform.social.metadata.model.MetadataType;
+
+public interface AttachmentService {
+
+  public static final MetadataType METADATA_TYPE = new MetadataType(7, "attachment");
```

Comment:
Please use plural since it's about a potential list of attached files for every object.

```suggestion
  public static final MetadataType METADATA_TYPE = new MetadataType(7, "attachments");
```

Reply:
done!